subprojects/docs/src/docs/userguide/api/kotlin_dsl.adoc

```diff
@@ -702,6 +702,55 @@ tasks.create("myTask") {
 // === `Property`, `Provider` and `NamedDomainObjectProvider`
 
 
+[[kotdsl:assignment]]
+== Kotlin Property assignment
+
+Gradle's Kotlin DSL supports Property assignment as an experimental feature that can be enabled by adding `systemProp.org.gradle.unsafe.kotlin.assignment=true` to the `gradle.properties` file.
+Property assignment reduces the verbosity for Kotlin DSL when <<lazy_configuration#lazy_properties,lazy properties>> are used.
+Property assignment works for properties that are _final_ and have type _Property_ or _ConfigurableFileCollection_.
```

Comment:
:x: 
```suggestion
Property assignment works for properties that are `final` and have type `Property` or `ConfigurableFileCollection`.
```

Maybe add that it's best to declare these properties via abstract getters, which makes sure about the `final` part?